File: app/src/main/java/com/travlog/android/apps/TravlogApplication.java
package com.travlog.android.apps;

import android.app.Application;
import android.content.Context;
import android.support.multidex.MultiDex;

import com.crashlytics.android.Crashlytics;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.facebook.stetho.Stetho;
import com.travlog.android.apps.libs.db.realm.RealmHelper;
import com.uphyca.stetho_realm.RealmInspectorModulesProvider;

import io.fabric.sdk.android.Fabric;
import timber.log.Timber;

public class TravlogApplication extends Application {

    private ApplicationComponent component;

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);

        MultiDex.install(this);
    }

    @Override
    public void onCreate() {
        super.onCreate();
        Fabric.with(this, new Crashlytics());

        Timber.plant(new Timber.DebugTree());

        this.component = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();
        component().inject(this);

        AppEventsLogger.activateApp(this);

        RealmHelper.getInstance().init(this);

//        Stetho.initialize(
//                Stetho.newInitializerBuilder(this)
//                        .enableDumpapp(Stetho.defaultDumperPluginsProvider(this))
//                        .enableWebKitInspector(RealmInspectorModulesProvider.builder(this).build())
//                        .build());
        Stetho.initializeWithDefaults(this);
    }

    public ApplicationComponent component() {
        return this.component;
    }
}

File: app/src/main/java/com/travlog/android/apps/TravlogApplication.kt
/*
 * Copyright 2019 Travlog. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.travlog.android.apps

import android.app.Application
import com.crashlytics.android.Crashlytics
import com.facebook.appevents.AppEventsLogger
import com.facebook.stetho.Stetho
import com.travlog.android.apps.libs.db.realm.RealmHelper
import io.fabric.sdk.android.Fabric
import timber.log.Timber

class TravlogApplication : Application() {

    private lateinit var component: ApplicationComponent

    override fun onCreate() {
        super.onCreate()
        Fabric.with(this, Crashlytics())

        Timber.plant(Timber.DebugTree())

        this.component = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()
        component().inject(this)

        Crashlytics.setString("git-sha", component.environment().build.sha())
        AppEventsLogger.activateApp(this)

        RealmHelper.init(this)

        //        Stetho.initialize(
        //                Stetho.newInitializerBuilder(this)
        //                        .enableDumpapp(Stetho.defaultDumperPluginsProvider(this))
        //                        .enableWebKitInspector(RealmInspectorModulesProvider.builder(this).build())
        //                        .build());
        Stetho.initializeWithDefaults(this)
    }

    fun component(): ApplicationComponent {
        return this.component
    }
}
